Issues linking with LLVM 18 in 1.13.9.0

### Problem

**Expected behavior:**  Build OSL library with LLVM 18

**Actual behavior:**  oslc fails to link with undefined references such as:

`/usr/bin/ld: ../../lib/liboslcomp.so.1.13.9: undefined reference to clang::api_notes::APINotesManager::APINotesManager(clang::SourceManager&, clang::LangOptions const&)'
`
We might just need -lclangAPINotes with LLVM 18

### Steps to Reproduce

1. `cmake .. -DCMAKE_CXX_STANDARD=17`
2. `make -j8`

### Versions

* OSL branch/version: 1.13.9.0 release
* OS: Ubuntu 24.04 LTS
* C++ compiler: clang 18.1.3 (Ubuntu 24.04 package)
* LLVM version: 18.0 (Ubuntu 24.04 package)
* OIIO version: 2.4.17 (Ubuntu 24.04 package)

--- a/src/cmake/modules/FindLLVM.cmake
+++ b/src/cmake/modules/FindLLVM.cmake
@@ -109,7 +109,7 @@ endif ()
 foreach (COMPONENT clangFrontend clangDriver clangSerialization
                    clangParse clangSema clangAnalysis clangAST
                    clangASTMatchers clangBasic clangEdit clangLex
-                   clangSupport)
+                   clangSupport clangAPINotes)
     find_library ( _CLANG_${COMPONENT}_LIBRARY
                   NAMES ${COMPONENT}
                   PATHS ${LLVM_LIB_DIR})

